### PR TITLE
[CAZB-2867] Add Payment in progress screen

### PR DESCRIPTION
**Pull request by tkaczu**
## Motivation and Context
https://eaflood.atlassian.net/browse/CAZB-2867

## Description
Add payer details to in progress page when somebody from organisation pays for the same CAZ

## How Has This Been Tested?
Manually, proper test cases updated and new added.

## Checklist:
- [ ] It contains only changes required by issue (does not contain other PR)
- [ ] Includes link to an issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/controllers/payments/payments_controller.rb [hunk 1]
```diff
@@ -239,7 +241,9 @@ def post_payment_details
     #   GET /payments/in_progress
     #
     def in_progress
-      # renders static page
+      api_response = AccountsApi.user(account_id: caz_lock_account_id, account_user_id: caz_lock_user_id)
+      @user = UsersManagement::User.new(api_response)
```
Comment (likeuwill) [inline app/controllers/payments/payments_controller.rb:245]: we have an email in redis, just fetch it.

File: features/step_definitions/payments_steps.rb [hunk 2]
```diff
@@ -102,13 +103,32 @@
...
+  mock_second_user_details
+end
+
+Then('I should be on the payment in progress page') do
+  expect_path(in_progress_payments_path)
+end
+
+Then('I should be on the dashboard page') do
+  expect_path(dashboard_path)
+end
```
Comment (likeuwill) [inline features/step_definitions/payments_steps.rb:122]: we already have this method
```
I should be on the Dashboard page
```

File: features/support/mock_users.rb [hunk 3]
```diff
@@ -33,6 +33,17 @@ def mock_user_details
     allow(AccountsApi).to receive(:user).and_return(read_response('users_management/user.json'))
   end
 
+  def mock_second_user_details
+    api_response = {
+      name: 'Mary Smith',
+      email: 'second_user@email.com',
+      owner: true,
```
Comment (likeuwill) [inline features/support/mock_users.rb:40]: we dont `owner`, please remove it

**Review by likeuwill — CHANGES_REQUESTED** [review]

File: app/controllers/payments/payments_controller.rb [hunk 4]
```diff
@@ -241,8 +241,8 @@ def post_payment_details
     #   GET /payments/in_progress
     #
     def in_progress
-      api_response = AccountsApi.user(account_id: caz_lock_account_id, account_user_id: caz_lock_user_id)
-      @user = UsersManagement::User.new(api_response)
+      # api_response = AccountsApi.user(account_id: caz_lock_account_id, account_user_id: caz_lock_user_id)
```
Comment (likeuwill) [inline app/controllers/payments/payments_controller.rb:244]: no need

File: spec/support/add_to_redis.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+module AddToRedis
+  def add_caz_lock_to_redis(user_id)
+    REDIS.hmset(caz_lock_key,
+                'account_id', user.account_id,
+                'user_id', user_id,
+                'caz_id', caz_id,
+                'email', user.email)
+  end
+end
```
Comment (likeuwill) [inline spec/support/add_to_redis.rb:11]: this helper will use only one time, please move it to rspec in private section or just in before block

**Review by likeuwill — CHANGES_REQUESTED** [review]

File: spec/support/add_to_redis.rb (hunk 5 above)
Comment (tkaczu) [reply spec/support/add_to_redis.rb:11]: It is used three times in two files - that's why I created this helper, please check:
`spec/requests/payments/payments/in_progress_spec.rb`and `spec/requests/payments/payments/local_authority_spec.rb`

**Review by likeuwill — APPROVED** [review]

**Pull request merged**